apps/examples/security_test/seclink/sl_auth_test.c

```diff
@@ -0,0 +1,900 @@
+	0xcb, 0xdd, 0x7d, 0x90, 0xc4, 0xbd, 0x70, 0x98, 0x48, 0x8e, 0x9c, 0x21, 0x9a,
+	0x73, 0x72, 0x4e, 0xff, 0xd6, 0xfa, 0xe5, 0x64, 0x47, 0x38, 0xfa, 0xa3, 0x1a,
+	0x4f, 0xf5, 0x5b, 0xcc, 0xc0, 0xa1, 0x51, 0xaf, 0x5f, 0x0d, 0xc8, 0xb4, 0xbd,
+	0x45, 0xbf, 0x37, 0xdf, 0x36, 0x5c, 0x1a, 0x65, 0xe6, 0x8c, 0xfd, 0xa7, 0x6d,
+	0x4d, 0xa7, 0x08, 0xdf, 0x1f, 0xb2, 0xbc, 0x2e, 0x4a, 0x43, 0x71
+};
+unsigned char g_buf_1024[] = {
+	0xa4, 0xd1, 0xcb, 0xd5, 0xc3, 0xfd, 0x34, 0x12, 0x67, 0x65, 0xa4, 0x42, 0xef,
+	0xb9, 0x99, 0x05, 0xf8, 0x10, 0x4d, 0xd2, 0x58, 0xac, 0x50, 0x7f, 0xd6, 0x40,
+	0x6c, 0xff, 0x14, 0x26,	0x6d, 0x31, 0x26, 0x6f, 0xea, 0x1e, 0x5c, 0x41, 0x56,
+	0x4b, 0x77, 0x7e, 0x69, 0x0f, 0x55, 0x04, 0xf2, 0x13, 0x16, 0x02, 0x17, 0xb4,
+	0xb0, 0x1b, 0x88, 0x6a, 0x5e, 0x91, 0x54, 0x7f,	0x9e, 0x27, 0x49, 0xf4, 0xd7,
+	0xfb, 0xd7, 0xd3, 0xb9, 0xa9, 0x2e, 0xe1, 0x90, 0x9d, 0x0d, 0x22, 0x63, 0xf8,
+	0x0a, 0x76, 0xa6, 0xa2, 0x4c, 0x08, 0x7a, 0x09, 0x1f, 0x53, 0x1d, 0xbf,	0x0a,
+	0x01, 0x69, 0xb6, 0xa2, 0x8a, 0xd6, 0x62, 0xa4, 0xd1, 0x8e, 0x73, 0xaf, 0xa3,
+	0x2d, 0x77, 0x9d, 0x59, 0x18, 0xd0, 0x8b, 0xc8, 0x85, 0x8f, 0x4d, 0xce, 0xf9,
+	0x7c, 0x2a, 0x24, 0x85, 0x5e, 0x6e, 0xeb, 0x22, 0xb3, 0xb2, 0xe5
+};
+
+TEST_SETUP(dh_generate_param)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ(SECLINK_OK, sl_init(&g_hnd));
+
+	g_dh_data.mode = HAL_DH_1024;
+	g_dh_data.G = (hal_data *)zalloc(sizeof(hal_data));
+	ST_EXPECT_NEQ(NULL, g_dh_data.G);
+	g_dh_data.G->data = g_buf_1024;
+	g_dh_data.G->data_len = sizeof(g_buf_1024);
+
+	g_dh_data.P = (hal_data *)zalloc(sizeof(hal_data));
+	ST_EXPECT_NEQ(NULL, g_dh_data.P);
+	g_dh_data.P->data = p_buf_1024;
+	g_dh_data.P->data_len = sizeof(p_buf_1024);
+
+	g_dh_data.pubkey = (hal_data *)zalloc(sizeof(hal_data));
+	ST_EXPECT_NEQ(NULL, g_dh_data.pubkey);
+	ST_EXPECT_EQ(0, sl_test_malloc_buffer(g_dh_data.pubkey, SL_TEST_AUTH_MEM_SIZE));
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(dh_generate_param)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ(SECLINK_OK, sl_deinit(g_hnd));
+
+	free(g_dh_data.G);
+	free(g_dh_data.P);
+	sl_test_free_buffer(g_dh_data.pubkey);
+	free(g_dh_data.pubkey);
+
+	ST_END_TEST;
+}
+
+TEST_F(dh_generate_param)
+{
+	ST_START_TEST;
+
+	hal_result_e hres = HAL_FAIL;
+	ST_EXPECT_EQ(SECLINK_OK, sl_dh_generate_param(g_hnd, SL_TEST_DH_X_SLOT, &g_dh_data, &hres));
+	ST_EXPECT_EQ(HAL_SUCCESS, hres);
+
+	ST_END_TEST;
+}
+
+/*
+ * Desc: Compute DH shared secret
+ * Refered https://developer.artik.io/documentation/security-api/see-authentication-test_8c-example.html
+ */
+static hal_data g_dh_shared_secret;
+
+TEST_SETUP(dh_compute_shared_secret)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ(SECLINK_OK, sl_init(&g_hnd));
+
+	g_dh_data.mode = HAL_DH_1024;
+
+	g_dh_data.G = (hal_data *)zalloc(sizeof(hal_data));
+	ST_EXPECT_NEQ(NULL, g_dh_data.G);
+	g_dh_data.G->data = g_buf_1024;
+	g_dh_data.G->data_len = sizeof(g_buf_1024);
+
+	g_dh_data.P = (hal_data *)zalloc(sizeof(hal_data));
+	ST_EXPECT_NEQ(NULL, g_dh_data.P);
```

Comment:
Is it okay not to free the `g_dh_data.G`?

Reply:
it's test case. so I think strict coding is not required when it failed.